internal/providers/docker/compose/project.go

```diff
@@ -20,12 +21,24 @@ import (
 func Parse(r io.Reader) (*Compose, error) {
 	dec := yaml.NewDecoder(r,
 		yaml.DisallowDuplicateKey(),
-		yaml.DisallowUnknownField(), // TODO: Handle this more gracefully.
 	)
 	var comp Compose
 	if err := dec.Decode(&comp); err != nil {
 		return nil, err
 	}
+
+	// Validate.
+	for key := range comp.Raw {
+		switch key {
+		case "version", "services", "networks", "volumes", "configs", "secrets":
+			// Ok.
+		default:
+			if !strings.HasPrefix(key, "x-") {
+				return nil, fmt.Errorf("unsupported top-level key in compose file: %q", key)
```

Comment:
What about doing this recursively for non-top-level keys?

Reply:
Yeah, it looks like compose 3.4 only specified extension fields at the top level, but the new compose spec allows them to appear as the keys of any object. Since that is the case, I'll do this recursively.